music21/musicxml/m21ToXml.py

```diff
@@ -2874,23 +2882,11 @@ def parseOneElement(self, obj):
         if len(obj.duration.dotGroups) > 1:
             obj.duration.splitDotGroups(inPlace=True)
 
-        # split at durations, if not a full-measure rest (e.g. whole rest in 9/8)
-        if 'GeneralNote' in classes and obj.duration.type == 'complex' and not (
-            'Rest' in classes and (
-                obj.fullMeasure in (True, 'always')
-                or (obj.fullMeasure == 'auto' and obj.duration == self.stream.barDuration)
-            )
-        ):
-            objList = obj.splitAtDurations()
-        else:
-            objList = [obj]
-
```

Comment:
I think both this and the 'inexpresible' bit above show that there is still a need to call splitAtDurations() in these edge cases. I'll restore the `else: objList = [obj]` bits so that all we skip here is the full-measure rest logic.

Reply:
For anyone else: the reason this was moved into splitAtDurations() was because it needed to be moved somewhere earlier during musicxml parsing so that beams could be made afterward (fixupNotationMeasured), and it seemed convenient to just make it splitAtDuration()'s responsibility.